### ISSUE TITLE
Ratings for the products

### ISSUE THREAD

**Issue by UO277653** [Arquisoft/dede_en2b#132]
We will add the possibility to write reviews for products

**Comment by UO277653** [comment]
Hello, @pglez82 , I have a small problem when loading the reviews (but I think this can be applied to anything that involves dynamically adding content using hooks). If I have, for example, 2 reviews for a product, when I enter the product page those 2 should appear, but sometimes (not even always) what happens is that they appear duplicated, so I get 4.

I spent some time debugging, and one possibility which was that the useEffect was called twice was discarded (I put a console.log to test it), so maybe it has to do with the setState( )? But I am not quite sure and at this point I am also a bit confused.

**Comment by UO277653** [comment]
Some more details that could help with reproducing the error
Branch: reviews
Problem: in the page with the products, when going to the details of one product sometimes the reviews are duplicated (for example in salmon by default it has 2 reviews, but if you go out of the page and enter again maybe 4 appear)

In order to run the app with the products, xampp has to be running with mysql (and a database named dede that can be empty), then the springboot folder has to be run as well and the app in the webapp folder with the npm start

**Comment by pglez82** [comment]
How do I run the spring bot part?

**Comment by UO277653** [comment]
Hello, in order to run the restapi what I do is, fron IntelliJ, File > Open, select the springboot folder inside the project

![imagen](https://user-images.githubusercontent.com/79276201/164628158-b57c6082-c558-493b-beee-fc15e0546367.png)

Open it on another window, and then run this class

![imagen](https://user-images.githubusercontent.com/79276201/164628253-24ae2631-a921-4179-8272-4ef95cbbd991.png)

Sometimes when loading the restapi it may happen that all the files have a lot of errors, this is due to something with the cache but it does not affect the execution

**Comment by pglez82** [comment]
I do not have intellij, nor xampp. 

I have created a mysql database using docker and I am running the spring bot application with mvn spring-boot:run.

It seems to be working. Let me see if I can reproduce the error in my computer.

**Comment by pglez82** [comment]
I am not able to reproduce the error. I have added multiple reviews and they are added only once. The only error that I found is that if press a second time the button "send", it changes the stats to two and half always. But I added many reviews and I've never seen them duplicate.

Note: I am doing the reviews as guest as I was not able to log in using my solid community pod. It redirects to the solid page and back to your site but I am still not logged in.
